### DISCUSSION TITLE
Is the best way of adding a background to the wrapper or canvases through Javascript?

### DISCUSSION THREAD

**Discussion by germancasares** [katspaugh/wavesurfer.js#3175] [Q&A] [answered]
Hi!

I'm trying to apply a background that moves alongside the scroll and I have been researching a bit and there are 2 ways of accomplishing that

1. With CSS write a selector like ::part(wrapper) or ::part(canvases) https://github.com/katspaugh/wavesurfer.js/issues/2900
    Although the wrapper or canvases elements don't have parts because of https://github.com/katspaugh/wavesurfer.js/pull/2907#discussion_r1233215179
2. With Javascript get the wrapper with getWrapper and add styles. https://github.com/katspaugh/wavesurfer.js/issues/2933#issuecomment-1605391003

So is my only option right now to apply CSS styles through Javascript like in option 2?

**Comment by katspaugh** [comment]
`getWrapper()` is meant for plugins only. Depending on what you mean by a background that moves alongside the scroll, I would apply CSS directly to your outer container.

**Reply by germancasares** [reply]
So it is really subtle, but here is a little recording. The "grid" background doesn't move in the first recording, but it does on the second.

I'm fine with adding the styles directly do the .wrapper via Javascript, I don't want to introduce more bugs later on with exposing the wrapper as a part, but was wondering what were my options.

With the background applied to the outer container:

https://github.com/katspaugh/wavesurfer.js/assets/18675185/e57a7d7e-7948-423c-ab25-089799c4afb4

With the background applied to the .wrapper or .canvases elements:

https://github.com/katspaugh/wavesurfer.js/assets/18675185/c9039e81-3795-4910-a75c-2c7344396eb4

**Reply by katspaugh** [reply, marked as answer]
I see. I think this is a legit use-case for custom styling of the wrapper. So feel free to make a PR to add a `part` attribute to it.

**Reply by germancasares** [reply]
Thank you! https://github.com/katspaugh/wavesurfer.js/pull/3182
